lib/spack/spack/installer.py

```diff
@@ -547,8 +547,8 @@ def log(pkg):
         log_file = os.path.join(os.path.dirname(packages_dir), log_file)
         fs.install(phase_log, log_file)
 
-    # Archive the environment used for the build
-    fs.install(pkg.env_path, pkg.install_env_path)
+    # Archive the environment modifications for the build.
+    fs.install(pkg.env_mods_path, pkg.install_env_path)
```

Comment:
okay maybe this answers my question if this pkg.env_mods_path is the one we saw passed earlier (the clean one)